homeassistant/components/telegram_bot/__init__.py

```diff
@@ -380,31 +388,62 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     if not domain_config:
         return False
 
+    # create config entries from YAML
+    for config_import in domain_config:
+        trusted_networks: list[IPv4Network] = config_import.get(
+            CONF_TRUSTED_NETWORKS, []
+        )
+        trusted_networks_str: list[str] = (
+            [str(trusted_network) for trusted_network in trusted_networks]
+            if trusted_networks
+            else []
+        )
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
```

Comment:
We should also create a repair issue to ask the user to remove the YAML config. Search for other integrations that import YAML to a config flow for examples.